quickwit-search/src/search_stream/leaf.rs

```diff
@@ -140,75 +140,224 @@ async fn leaf_search_stream_single_split(
     warmup(
         &*searcher,
         query.as_ref(),
-        &fast_field_collector_builder.fast_field_to_warm(),
+        &request_fields.fast_fields_for_request(), // FIXME: pass the right vec.
     )
     .await?;
-    let collect_handle = spawn_blocking(move || {
-        collect_fast_field_values(
-            &fast_field_collector_builder,
-            &searcher,
-            query,
-            output_format,
-        )
-    });
+
     let span = info_span!(
         "collect_fast_field",
         split_id = %split.split_id,
-        fast_field=%fast_field_to_extract,
+        request_fields=%request_fields,
     );
+
     let _ = span.enter();
+    let m_request_fields = request_fields.clone();
+    let collect_handle = spawn_blocking(move || {
+        let mut buffer = Vec::new();
+        match m_request_fields.fast_field_types() {
+            (Type::I64, None) => {
+                let collected_values = collect_values::<u64>(
+                    &m_request_fields,
+                    stream_request.start_timestamp,
+                    stream_request.end_timestamp,
+                    searcher,
+                    query.as_ref(),
+                )?;
+                super::serialize::<u64>(&collected_values, &mut buffer, output_format).map_err(
```

Comment:
```suggestion
                super::serialize::<i64>(&collected_values, &mut buffer, output_format).map_err(
```

the weird results are coming from this mistake ;)